langchain/src/document_loaders/fs/text.ts

```diff
@@ -31,19 +31,32 @@ export class TextLoader extends BaseDocumentLoader {
         );
       }
     });
-    return parsed.map(
-      (pageContent, i) =>
-        new Document({
-          pageContent,
+
+    return parsed.map((parsedDoc, i) => {
+      if (typeof parsedDoc === "string") {
+        return new Document({
+          pageContent: parsedDoc,
           metadata:
             parsed.length === 1
               ? metadata
               : {
                   ...metadata,
                   line: i + 1,
                 },
-        })
-    );
+        });
+      }
+      return new Document({
+        ...parsedDoc,
+        metadata:
+          parsed.length === 1
+            ? { ...metadata, ...parsedDoc.metadata }
+            : {
+                ...metadata,
+                line: i + 1,
```

Comment:
I suppose it depends on whether or not we want to allow the parser to override that. I originally did this because some of the vector databases have limits on how large your metadata can be. So maybe you want to create a custom parser that overrides it for your own usage? I can just switch it up though.

Reply:
One more thing is the meaning of `line` at the moment this just increments it based on the string line. In markdown, where we are breaking it into sections, we can either set the `line` to the actual line in the md file, or leave it as is which just increments per section provided.

Reply:
Makes sense, let's keep it that way 👍